sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/implementation/models/AttestationTokenImpl.java

```diff
@@ -0,0 +1,343 @@
+     */
+    @Override public AttestationSigner getCertificateChain() {
+        if (jwsHeader != null) {
+            List<Base64> certChain = jwsHeader.getX509CertChain();
+            return AttestationSignerImpl.fromCertificateChain(certChain);
+        }
+        return null;
+    }
+
+    /**
+     * Returns a URI which can be used to retrieve a JSON Web Key which can verify the signature on
+     * this token.
+     * @return URI at which a JWK can be retrieved.
+     */
+    @Override public URI getJsonWebKeyUrl() {
+        return jwsHeader != null ? jwsHeader.getJWKURL() : null;
+    }
+
+    /**
+     * Returns the signer for this token if the caller provided a JSON Web Key.
+     * @return Attestation signer representing the signer of the token.
+     */
+    @Override public AttestationSigner getJsonWebKey() {
+        if (jwsHeader != null) {
+            JWK jwk = jwsHeader.getJWK();
+            if (jwk == null) {
+                return null;
+            }
+            return AttestationSignerImpl.fromJWK(jwk);
+        }
+        return null;
+    }
+
+    /**
+     * Returns the SHA-256 thumbprint of the leaf certificate in the getCertificateChain.
+     * @return the SHA-256 thumbprint of the leaf certificate returned by getCertificateChain.
+     */
+    @Override public byte[] getSha256Thumbprint() {
+        return jwsHeader != null ? jwsHeader.getX509CertSHA256Thumbprint().decode() : null;
+    }
+
+    /**
+     * Returns the SHA-1 thumbprint of the leaf certificate in the getCertificateChain.
+     * @return the SHA-1 thumbprint of the leaf certificate returned by getCertificateChain.
+     */
+    @Override public byte[] getThumbprint() {
+        return jwsHeader != null ? jwsHeader.getX509CertThumbprint().decode() : null;
+    }
+
+    /**
+     * Returns a URI which can be used to retrieve an X.509 certificate which can verify the signature
+     * on this token.
+     * @return URI at which an X.509 certificate can be retrieved.
+     */
+    @Override public URI getX509Url() {
+        return jwsHeader != null ? jwsHeader.getX509CertURL() : null;
+    }
+
+    @Override
+    public String[] getCritical() {
+        return jwsHeader.getCriticalParams().toArray(new String[0]);
+    }
+
+    @Override
+    public String getType() {
+        return jwsHeader.getType().getType();
+    }
+
+    @Override
+    public String getContentType() {
+        return jwsHeader.getContentType();
+    }
+
+    /**
+     * Create an unsecured attestation token with an empty body. Used for resetting attestation
+     * policies.
+     * @return Newly created unsecured attestation token with an empty body.
+     */
+    public static AttestationToken createUnsecuredToken() {
+        // Create a AttestationToken using the well known unsecured JWT header.
+        // See <a href='https://datatracker.ietf.org/doc/html/rfc7519#section-6.1' RFC 7519 section 6.1/>.
+        return new AttestationTokenImpl("eyJhbGciOiJub25lIn0..");
```

Comment:
Should this be a const string?

Reply:
I'm confused - aren't manifest constants in Java constant values?  How would I say "this is a constant string"?

Or are you saying I should do the logical equivalent of:
```java
const int two = 2;

callMethod(two);
```

Because that's basically what saying:
```java

const String emptyAttestationToken = "eyJhbGciOiJub25lIn0..";
return new AttestationTokenImpl(emptyAttestationToken);
```
does.

Reply:
final String is what you would essentially use and I meant the logical equivalent of your example  - I have found us mostly using that convention across all SDK. Yes, semantically they are the same, so I will leave the choice to you :)

Reply:
Java string constants aren't static? That's fascinating. But that makes sense.

Reply:
QQ: What is a MetricsAdvisorServiceVersion? That appears to be an enumeration listing service versions? Do you have examples from the training which show how to set up such an enumeration?

Reply:
Yes, that is basically the enum that has the different versions of the SDK and internally maps them to the different swagger versions. I quickly looked at the tutorial repo to see if we have any tutorial on that but did not find it, I will ask around just in case :)

Reply:
I reverse engineered what MetricsAdvisor is doing, but there's one construct I cannot understand...

```java
public enum MetricsAdvisorServiceVersion implements ServiceVersion {
    V1_0("v1.0");


    private final String version;

```
From [https://github.com/Azure/azure-sdk-for-java/blob/a804ef19f7f7d099747bd60d67d4b9bf079e2ead/sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/MetricsAdvisorServiceVersion.java#L11-L15](this code).

Any ideas what that does and/or how it works?

Reply:
Ok, there must have been a C&P error - I copied the entire file from MetricsAdvisor and it compiled.